File: services/delivery/src/main/java/io/wkrzywiec/fooddelivery/delivery/domain/incoming/OrderCanceled.java
package io.wkrzywiec.fooddelivery.delivery.domain.incoming;

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody;

import java.util.UUID;

public record OrderCanceled(UUID orderId, int version, String reason) implements IntegrationMessageBody {
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/OrderCanceled.kt
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody
import java.util.*

data class OrderCanceled(val id: UUID, val aggregateVersion: Int, val reason: String) : IntegrationMessageBody {
    override fun orderId(): UUID = id
    override fun version(): Int = aggregateVersion
}
